src/GafferArnoldUI/GoboVisualiser.cpp

```diff
@@ -117,7 +120,12 @@ CompoundDataPtr getter( const OSLTextureCacheGetterKey &key, size_t &cost )
 {
 	// make the cost be image data in bytes
 	cost = key.resolution * key.resolution * 3 * 4;
-	return GafferOSL::ShadingEngineAlgo::shadeUVTexture( key.shaderNetwork, V2i( key.resolution ), key.output );
+
+	if( ShaderNetworkPtr textureNetwork = VisualiserAlgo::conformToOSLNetwork( key.output, key.shaderNetwork ) )
+	{
+		return GafferOSL::ShadingEngineAlgo::shadeUVTexture( textureNetwork.get(), V2i( key.resolution ) );
+	}
+	return nullptr;
```

Comment:
We need to deal with this null return in `GoboVisualiser::visualise()`. It currently causes a crash if you connect an unsupported shader into the gobo.

```
*  1 0x00007f01fd8135ef [libpthread.so.0     ] _L_unlock_13                                                                                                                                                                                                                          [funlockfile.c     :  ?]
>> 2 0x00007f0177f505f3 [libGafferArnoldUI.so] (anonymous namespace)                                                                                                                                                                                                                 [GoboVisualiser.cpp:  ?]
*  3 0x00007f0177f505f3 [libGafferArnoldUI.so]                                                                                                                                                                                                                                       [GoboVisualiser.cpp:  ?]
*  4 0x00007f01cd4affeb [libGafferScene.so   ] (anonymous namespace)                                                                                                                                                                                                                 [Renderer.cpp      :  ?]
*  5 0x00007f01cd51e8f4 [libGafferScene.so   ] GafferScene::RenderController::SceneGraph::attributesInterface(IECoreScenePreview::Renderer*)                                                                                                                                         
*  6 0x00007f01cd5230cb [libGafferScene.so   ] GafferScene::RenderController::SceneGraph::update(std::vector<IECore::InternedString, std::allocator<IECore::InternedString> > const&, unsigned int, GafferScene::RenderController::SceneGraph::Type, GafferScene::RenderController*) 
*  7 0x00007f01cd523463 [libGafferScene.so   ] GafferScene::RenderController::SceneGraphUpdateTask::execute()                                    
```

Reply:
Thanks for catching that John. https://github.com/GafferHQ/gaffer/pull/3667/commits/72203ea6602116ae321a78422858a7ceabc54c82